registry.py

```diff
@@ -1000,6 +998,36 @@ def elasticsearch(serializer, catalog):
                 }
             }
         }
+    # Heatmap support.
+    if a_hm_limit:
+        # by default is q_geo.
+        heatmap_filter_box = [[Xmin, Ymax], [Xmax, Ymin]]
+        # but if user sends the hm filter:
+        if a_hm_filter:
+            a_hm_filter = str(a_hm_filter)[1:-1]
+            Ymin, Xmin = a_hm_filter.split(" TO ")[0].split(",")
+            Ymax, Xmax = a_hm_filter.split(" TO ")[1].split(",")
+            heatmap_filter_box = [[Xmin, Ymax], [Xmax, Ymin]]
+
+        heatmap = {
+            "heatmap": {
+                "field": "layer_geoshape",
+                "dist_err_pct": 0.07,
+                "geom": {
+                    "geo_shape": {
+                        "layer_geoshape": {
+                            "shape": {
+                                "type": "envelope",
+                                "coordinates": heatmap_filter_box
+                            },
+                            "relation": "within"
+                        }
+                    }
+                }
+            }
+        }
+        aggs_dic["viewport"] = heatmap
```

Comment:
The `viewport` variable does not seem to convey a lot of meaning, but since it is not passed back to the client `a.hm` is I think we can leave it.